airflow/cli/commands/task_command.py

```diff
@@ -404,7 +404,14 @@ def task_test(args, dag=None):
     # Add CLI provided task_params to task.params
     if args.task_params:
         passed_in_params = json.loads(args.task_params)
+        for k, v in task.params.items():
+            if k in passed_in_params:
+                v.default = passed_in_params.pop(k)
+                v()  # to raise if there are any validation issues
```

Comment:
This looks like it would fail for basic/non-validating param types

Reply:
We are converting any kind of param into `Param` object only, of course with no attached validations, so it'll work.